publication-rest/src/main/java/no/unit/nva/publication/uploadfile/restmodel/CompleteUploadRequestBody.java

```diff
@@ -0,0 +1,33 @@
+package no.unit.nva.publication.uploadfile.restmodel;
+
+import static java.util.Objects.requireNonNull;
+import com.amazonaws.services.s3.model.CompleteMultipartUploadRequest;
+import com.amazonaws.services.s3.model.PartETag;
+import java.util.List;
+import nva.commons.apigateway.exceptions.BadRequestException;
+
+public record CompleteUploadRequestBody(String uploadId, String key, List<CompleteUploadPart> parts) {
+
+    public List<PartETag> partETags() {
+        return parts().stream().filter(CompleteUploadPart::hasValue).map(CompleteUploadPart::toPartETag).toList();
+    }
+
+    public CompleteMultipartUploadRequest toCompleteMultipartUploadRequest(String bucketName) {
+        var completeMultipartUploadRequest = new CompleteMultipartUploadRequest();
+        completeMultipartUploadRequest.setBucketName(bucketName);
+        completeMultipartUploadRequest.setKey(key());
+        completeMultipartUploadRequest.setUploadId(uploadId());
+        completeMultipartUploadRequest.setPartETags(partETags());
+        return completeMultipartUploadRequest;
```

Comment:
In `CompleteUploadResponseBody`, you use the constructor with parameters, I guess I prefer a constructor, compared  to setters....

Reply:
Its aws class, I take in use "builder" instead:

```
    public CompleteMultipartUploadRequest toCompleteMultipartUploadRequest(String bucketName) {
        return new CompleteMultipartUploadRequest()
                   .withBucketName(bucketName)
                   .withKey(key())
                   .withUploadId(uploadId())
                   .withPartETags(partETags());
    }
```